lib/rules/import-notation/index.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const valueParser = require('postcss-value-parser');
+const setAtRuleParams = require('../../utils/setAtRuleParams');
+const getAtRuleParams = require('../../utils/getAtRuleParams');
+const atRuleParamIndex = require('../../utils/atRuleParamIndex');
+
+const ruleName = 'import-notation';
+
+const messages = ruleMessages(ruleName, {
+	expected: (importUrlOrString, primary) => `Expected "${importUrlOrString}" to be a "${primary}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/import-notation',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, _, context) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: ['string', 'url'],
+		});
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules('import', checkAtRuleImportParams);
```

Comment:
[suggestion] Better to support case-insensitive `@import`.

```suggestion
		root.walkAtRules(/^import$/i, checkAtRuleImportParams);
```